### ISSUE TITLE
DMA: Out of SW-IOMMU space for 9100 bytes at device 0000:05:07.0

### ISSUE THREAD

**Issue by weberkai** [lwfinger/rtl8190p#1]
Hi!

I have this NIC: http://www.encore-usa.com/br/taxonomy/term/66/all .
Ndiswrapper dont work, as I am using a 64 bit kernel.

This driver is awesome, but since I compiled my connection drops often and sometimes I need to reboot, or else it won't connect again.

I've found this link, but I couldn't adapt it for rtl8190p:
http://www.spinics.net/lists/linux-driver-devel/msg08658.html

I want to help improving this driver.

Please, can you send me some tips on how to traceback this issue? Trying gcc debug will help?
Thank you!

Best regards,
Weber Kai

**Comment by weberkai** [comment]
I've done some tests:
For every pci_map instruction I've added a printk("****_m1") and for every pci_unmap I've added a printk("**_**u1") changing only the number.
Then I count every instruction.
In log we have 250k map and 20k unmap.

In rtl_core.c in function rtl8192_rx the pci_map_single is called 200k times.
In the same function, the unmap is never called, as my kernel version is greater than 2.5.0

I think that's why we are getting out of iommu space.
